sycl/include/CL/sycl/usm/usm_allocator.hpp

```diff
@@ -121,7 +121,7 @@ class usm_allocator {
   // Deallocate memory
   void deallocate(pointer Ptr, size_t size) {
     if (Ptr) {
-      detail::usm::free(Ptr, mContext);
+      detail::usm::free(Ptr, *mContext);
```

Comment:
It's very bad idea to store pointer to sycl::context and sycl::device. What is the reason for that?

Reply:
Would you prefer a shared ptr?

Reply:
Actually right now you can't even initialize them after the fact, so we could change to a reference even and delete the default constructor.

Reply:
To be clear - this class needs to keep track of the context and device the allocator is associated with.

Reply:
Actually - since we're dealing with the sycl wrappers for these objects - would you prefer I store them by-value?